doc/modules/neighbors.rst

```diff
@@ -464,6 +464,21 @@ leaf nodes.  The level of this switch can be specified with the parameter
 
 ``leaf_size`` is not referenced for brute force queries.
 
+Valid Metrics for Nearest Neighbor Algorithms
+---------------------------------------------
+
+For a list of available metrics, see the documentation of the :class:`DistanceMetric`
+class.
+
+ A list of valid metrics for any of the above algorithms can be obtained by using their
+``valid_metric`` attribute. For example, valid metrics for ``KDTree`` can be generated by:
+
+    >>> from sklearn.neighbors import KDTree
+    >>> print(sorted(KDTree.valid_metrics))
+     ['chebyshev' 'cityblock' 'euclidean' 'infinity' 'l1' 'l2' 'manhattan'
+     'minkowski' 'p']
+
```

Comment:
To pass [the current failing test on the CI](https://dev.azure.com/scikit-learn/scikit-learn/_build/results?buildId=26542&view=logs&j=13650d7c-49e8-54f3-0598-c3480d1c1e4f&t=20813006-c7bd-527c-f387-55e46b5480d7&l=82):

```suggestion
    ['chebyshev', 'cityblock', 'euclidean', 'infinity', 'l1', 'l2', 'manhattan', 'minkowski', 'p']
```